src/duckling/project/project.service.ts

```diff
@@ -136,13 +139,31 @@ export class ProjectService {
      * @param key The key of the custom attribute
      * @param content The json that describes the attribute
      */
-    addCustomAttribute(key : string, content : any) {
+    addCustomAttribute(key : string, content : JsonSchema) {
         let currentAttributes = this._project.customAttributes;
         let newAttribute = {key, content };
         let newAttributes = currentAttributes ? currentAttributes.concat([newAttribute]) : [].concat([newAttribute]);
         this._storeService.dispatch(changeCustomAttributes(newAttributes));
     }
 
+    hasCustomAttribute(attributeKey : AttributeKey) : boolean {
```

Comment:
You should rename this isCustomAttribute. hasCustomAttribute sounds like a function that is checking to see if an entity has a custom attribute.